metapool/plate.py

```diff
@@ -412,3 +413,287 @@ def _autopool_norm(pool_failures, total_nmol, min_conc, sample_concs,
+        # counter '1' will become 2. If quad 2 is replicated three times,
+        # counter '1' will become 4. If a quadrant becomes a destination then
+        # this value will be overwritten. If empty, this value will be unused.
+        self.rep_counters = {'1': 1, '2': 1, '3': 1, '4': 1}
+
+        self.status = {'1': PlateReplication.STATUS_EMPTY,
+                       '2': PlateReplication.STATUS_EMPTY,
+                       '3': PlateReplication.STATUS_EMPTY,
+                       '4': PlateReplication.STATUS_EMPTY}
+
+        # this allows us to store output dataframes for each quadrant,
+        # overwrite them on demand, and save final concatenation for when
+        # we're done.
+        self.data = {'1': None, '2': None, '3': None, '4': None}
+
+    def _get_quadrant(self, quadrant):
+        d = OrderedDict()
+        for i in range(1, 9):
+            row_96 = PlateReplication.row_letters[i - 1]
+            if quadrant in ['1', '2']:
+                row_384 = PlateReplication.row_letters[(2 * i) - 2]
+            else:
+                row_384 = PlateReplication.row_letters[(2 * i) - 1]
+
+            for j in range(1, 13):
+                col_96 = j
+                if quadrant in ['1', '3']:
+                    col_384 = (2 * j) - 1
+                else:
+                    col_384 = (2 * j)
+
+                k = "%s%s" % (row_96, col_96)
+                v = "%s%s" % (row_384, col_384)
+                d[k] = v
+
+        return d
+
+    def get_384_well_location(self, well_96_id, quadrant):
+        '''
+        Translate a 96-well plate + a quadrant into a 384-well plate cell
+        :param well_96_id: A 96-well plate ID
+        :param quadrant: A quadrant of a 384-well plate e.g. '1', '2', '3', '4'
+        :return: A 384-well plate ID
+        '''
+        quadrant = str(quadrant)
+        if quadrant in self.map_to_384:
+            if well_96_id in self.map_to_384[quadrant]:
+                return self.map_to_384[quadrant][well_96_id]
+
+    def get_96_well_location_and_quadrant(self, well_384_id):
+        '''
+        Translate a 384-well plate ID to a 96-well plate ID and a quadrant
+        :param well_384_id: A 384-well plate ID
+        :return: A tuple of (quadrant, 96-well plate ID)
+        '''
+        # not an optimal search but mean-search-space is 192 and won't grow.
+        for quadrant in PlateReplication.quadrants:
+            for k in self.map_to_384[quadrant]:
+                if self.map_to_384[quadrant][k] == well_384_id:
+                    return quadrant, k
+
+    def _map_quadrants(self, src_quad, dst_quad):
+        # since all values in self.d are generated in the same order by
+        # the same function and saved as OrderedDicts, the 384-well locations
+        # from any src quadrant will map directly to those of the dst quad.
+        src = list(self.map_to_384[str(src_quad)].values())
+        dst = list(self.map_to_384[str(dst_quad)].values())
```

Comment:
I'm not sure the list conversion is necessary but it might be for map, do you know?